### PR TITLE
feat: aligned verification data stored locally stores cbor and json

### LINKED ISSUE #1263: Fix: aligned verification data stored locally should be json and not cbor
the file has .json extension but its content is actually CBOR

### PR DESCRIPTION
**Pull request by uri-99**
- The function was also moved to the SDK
- FileError was added for file-related errors
  - IO, Serialize

### CODE REVIEW

File: batcher/aligned-sdk/src/sdk.rs [hunk 1]
```diff
@@ -655,6 +660,81 @@ pub async fn get_balance_in_aligned(
...
+) -> Result<(), errors::FileError> {
+    let _ = save_response_cbor(
+        batch_inclusion_data_directory_path.clone(),
+        &aligned_verification_data.clone(),
+    )?;
+    let _ = save_response_json(
+        batch_inclusion_data_directory_path,
+        &aligned_verification_data,
+    )?;
+    Ok(())
```
Comment (IAvecilla) [inline batcher/aligned-sdk/src/sdk.rs]:
```suggestion
    save_response_cbor(
        batch_inclusion_data_directory_path.clone(),
        &aligned_verification_data.clone(),
    )?;
    save_response_json(
        batch_inclusion_data_directory_path,
        &aligned_verification_data,
    )
```

File: batcher/aligned-sdk/src/sdk.rs [hunk 2]
```diff
@@ -655,6 +660,81 @@ pub async fn get_balance_in_aligned(
...
+
+    let batch_inclusion_data_path =
+        batch_inclusion_data_directory_path.join(batch_inclusion_data_file_name);
+
+    let data = cbor_serialize(&aligned_verification_data)?;
+
+    let mut file = File::create(&batch_inclusion_data_path)
+        .map_err(|e| errors::FileError::IoError(batch_inclusion_data_path.clone(), e))?;
+    file.write_all(data.as_slice())
+        .map_err(|e| errors::FileError::IoError(batch_inclusion_data_path.clone(), e))?;
```
Comment (IAvecilla) [inline batcher/aligned-sdk/src/sdk.rs]: You can implement the `From<io::Error>` for the `FileError` and get rid of the `map_err` everytime.

File: batcher/aligned-sdk/src/sdk.rs [hunk 3]
```diff
@@ -655,6 +660,81 @@ pub async fn get_balance_in_aligned(
...
+
+    let batch_inclusion_data_path =
+        batch_inclusion_data_directory_path.join(batch_inclusion_data_file_name);
+
+    let data = serde_json::to_vec(&aligned_verification_data).unwrap();
+
+    let mut file = File::create(&batch_inclusion_data_path)
+        .map_err(|e| errors::FileError::IoError(batch_inclusion_data_path.clone(), e))?;
+    file.write_all(data.as_slice())
+        .map_err(|e| errors::FileError::IoError(batch_inclusion_data_path.clone(), e))?;
```
Comment (IAvecilla) [inline batcher/aligned-sdk/src/sdk.rs]: Same as mentioned above, with that change we can remove the error mapping here

File: batcher/aligned-sdk/src/sdk.rs [hunk 4]
```diff
@@ -655,6 +660,81 @@ pub async fn get_balance_in_aligned(
...
+    let batch_merkle_root = &hex::encode(aligned_verification_data.batch_merkle_root)[..8];
+    let batch_inclusion_data_file_name = batch_merkle_root.to_owned()
+        + "_"
+        + &aligned_verification_data.index_in_batch.to_string()
+        + ".json";
+
+    let batch_inclusion_data_path =
+        batch_inclusion_data_directory_path.join(batch_inclusion_data_file_name);
+
+    let data = serde_json::to_vec(&aligned_verification_data).unwrap();
```
Comment (IAvecilla) [inline batcher/aligned-sdk/src/sdk.rs]: Just out of curiosity, will this still saving the bytes directly into the json file? Wouldn't be good to add the encoding to hex before saving to make it easier to use?

**Review by IAvecilla — CHANGES_REQUESTED** [review]

File: batcher/aligned-sdk/src/sdk.rs (hunk 4 above)
Comment (uri-99) [reply batcher/aligned-sdk/src/sdk.rs]: This has shown me the `proof_commitment` stored is different than the `proof_hash` shown in the explorer. I am searching why and how to fix

Comment (uri-99) [reply batcher/aligned-sdk/src/sdk.rs]: Solved! Explorer was using Sha3 instead of Keccak (I thought they were always interchangeable, they are not.)

Comment (IAvecilla) [reply batcher/aligned-sdk/src/sdk.rs]: Excellent!

**Review by IAvecilla — APPROVED** [review]
LGTM!

**Review by PatStiles — APPROVED** [review]

**Review by entropidelic — APPROVED** [review]

**Pull request merged**